### PR TITLE
Create model for conviction entities

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-741

This is the first step of implementing conviction checking entirely in the engine instead of making calls to waste-carriers-service.

This PR sets up an Entity model to match the data we store in the database.

### CODE REVIEW

**Comment by irisfaraway** [comment]
Aside from the question of whether or not to exclude certain terms (see https://github.com/DEFRA/waste-carriers-service/blob/398abf7b35d953a3878d6490efb80f6b68b1cdb9/src/main/java/uk/gov/ea/wastecarrier/services/match/CompanyMatch.java#L24-L33) – this is ready for review.

**Comment by irisfaraway** [comment]
Resolved the question about the common terms on Slack, so bumping this back out of review to sort it.

File: app/models/waste_carriers_engine/convictions_check/entity.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+        where(name: /#{escaped_first_name}/i).and(name: /#{escaped_last_name}/i)
+      }
+
+      scope :matching_date_of_birth, lambda { |term|
+        where(date_of_birth: term)
+      }
+
+      scope :matching_company_number, lambda { |term|
+        escaped_term = ::Regexp.escape(term) if term.present?
```
Comment (cintamani) [inline app/models/waste_carriers_engine/convictions_check/entity.rb]: shall we return if there is no term present? Line 41 will fail if the term is not present and we are missing test coverage for when a term is empty

Comment (irisfaraway) [reply app/models/waste_carriers_engine/convictions_check/entity.rb]: Good shout – I've added test coverage and set the scopes to raise an error if the terms aren't present!

**Comment by irisfaraway** [comment]
Don't know why CodeClimate is stuck but this is ready for review again 🤷‍♀

**Comment by irisfaraway** [comment]
@cintamani Cool. I can crack on with other stuff in the meantime so let's sort it out and merge then :)
